bin/getInfoRippled.sh

```diff
@@ -1,41 +1,72 @@
 #!/usr/bin/env bash
 
-TMP_LOC=$(mktemp -d /tmp/ripple_info.XXXX)
-echo ${TMP_LOC}
+rippled_exe=/opt/ripple/bin/rippled
+conf_file=/etc/opt/ripple/rippled.cfg
 
-cd ${TMP_LOC}
+while getopts ":e:c:" opt; do
+    case $opt in
+        e)
+            rippled_exe=${OPTARG}
+            ;;
+        c)
+            conf_file=${OPTARG}
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG"
+    esac
+done
 
-# Send output from this script to a log file
-## this captures any messages
-## or errors from the script itself
+tmp_loc=$(mktemp -d /tmp/ripple_info.XXXX)
+cd /tmp
+chmod 751 ripple_info.*
+cd ~
+echo ${tmp_loc}
 
-LOG_FILE=${TMP_LOC}/get_info.log
-exec 3>&1 1>>${LOG_FILE} 2>&1
+cleaned_conf=${tmp_loc}/cleaned_rippled_cfg.txt
 
-## Send all stdout files to /tmp
+db=$(sed -r -e 's/\<s[a-zA-Z0-9]{28}\>/secretsecretsecretsecretmaybe/g' ${conf_file} |\
+            awk -v OUT_FILE=${cleaned_conf} '
+BEGIN {skip=0; db_path="";print > OUT_FILE}
+/^\[validation_seed\]/ {skip=1; next}
+/^\[node_seed\]/ {skip=1; next}
+/^\[.*\]/ {skip=0}
+skip==1 {next}
+save==1 {save=0;db_path=$0}
+/^\[database_path\]/ {save=1}
+{print >> OUT_FILE}
+END {print db_path}
+')
 
-/opt/ripple/bin/rippled server_info > ${TMP_LOC}/server_info.txt
-df -h                               > ${TMP_LOC}/free_disk_space.txt
-cat /proc/meminfo                   > ${TMP_LOC}/amount_mem.txt
-cat /proc/swaps                     > ${TMP_LOC}/swap_space.txt
-ulimit -a                           > ${TMP_LOC}/reported_current_limits.txt
+echo "database_path: ${db}"
+df ${db} > ${tmp_loc}/db_path_df.txt
+echo
 
-TMP_CONF=${TMP_LOC}/rippled.cfg
+# Send output from this script to a log file
+## this captures any messages
+## or errors from the script itself
 
-cp /etc/opt/ripple/rippled.cfg ${TMP_CONF}
+log_file=${tmp_loc}/get_info.log
+exec 3>&1 1>>${log_file} 2>&1
 
-SEED_LINE_NUM=$(grep -n "\[validation_seed\]" ${TMP_CONF} | tail -n 1 | awk -F ':' '{print $1}')
-SEED_LINE_NUM=$((SEED_LINE_NUM + 1))
+## Send all stdout files to /tmp
 
-awk -v SEED_LINE_NUM=${SEED_LINE_NUM} 'NR != SEED_LINE_NUM' "${TMP_CONF}" > ${TMP_LOC}/cleaned_rippled_cfg.txt
+${rippled_exe} server_info          > ${tmp_loc}/server_info.txt
```

Comment:
I don't know what the use case is going to be for this script, but I think we should be more careful about our scrubbing of the inputs. For example, `getInfoRippled.sh -e "rm -rf ~"`. This could be mitigated with `if [[ -x ${rippled_exe} ]]`.